stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -284,6 +284,22 @@ end
     mul!(C, A, J.λ, alpha, beta)
 @inline mul!(C::AbstractVecOrMat, J::UniformScaling, B::AbstractVecOrMat, alpha::Number, beta::Number) =
     mul!(C, J.λ, B, alpha, beta)
+
+function mul!(out::AbstractMatrix{T}, a::Bool, B::UniformScaling{T}, α::Bool, β::Bool) where {T}
+    if !β  # zero contribution of the out matrix
+        fill!(out,zero(T));
+    end
+    if (a*α)
+        # Add B.λ*I
+        m = min(size(out)...)
+        s = B.λ
+        @inbounds for i = 1:m;
+            out[i,i] += s;
+        end
+    end
+    return out
+end
```

Comment:
I find the method signature extremely narrow. How about relaxing the `Bool`s to `Number`s. I don't think this makes it more expensive for the Bool-case.
```suggestion
function mul!(out::AbstractMatrix{T}, a::Number, B::UniformScaling, α::Number, β::Number) where {T}
    checksquare(out)
    if iszero(β)  # zero contribution of the out matrix
        fill!(out, zero(T))
    elseif !isone(β)
        rmul!(out, β)
    end
    s = convert(T, a*B.λ*α)
    if !iszero(s)
        @inbounds for i in diagind(out)
            out[i] += s
        end
    end
    return out
end
```
In particular, this includes a square-check that I believe we would want, otherwise nothing stops the code from plugging the `UniformScaling` into a rectangular matrix.

Reply:
Thanks. Good point about checksquare. Your suggestion will throw and `InexactError` if the user gives incompatible types. With this added functionality, it makes more sense to include in `mul!` (and not in `axpby!`).

Reply:
Right, but that's how it always works with inplace methods: we convert to the target type, and if that's impossible, we get a conversion error.

Reply:
Your change from `A[i,i]` to `A[i]` with `i in diagind`: Is this better in general? In `uniformscaling.jl` you have both versions in different functions.

Reply:
Yeah, I don't know, we could as well use the `intersect` construct. `diagind` always returns a range, and that should be just right for matrices with linear indexing. I guess you could benchmark a little and see if it makes any difference for huge matrices.

Reply:
No difference noticed.